### PR TITLE
Clarifying when salt needs to be specified in hashed password object

### PR DESCRIPTION
**Pull request by motikagan-okta**
## Description:
- **What's changed?** Instead of saying that the `salt` property of the Hashed Password object is optional, say that it is required only for the specific algorithms that use a salt.
- **Is this PR related to a Monolith release?** No

### CODE REVIEW

**Comment by motikagan-okta** [comment]
Screen Shot 2020-04-07 at 2 49 27 PM

**Comment by motikagan-okta** [comment]
@rameshbhojan-okta, could you please also confirm that the `Min Value` and `Max Value` columns for the `salt` property make sense.

**Comment by motikagan-okta** [comment]
Spoke to @rameshbhojan-okta out-of-band. Since the `Min Value` and `Max Value` columns actually mean minimum and maximum length here, not value, but do mean min and max value for the `workFactor` property, seems best to delete those two column and put the information in the descriptions of the properties.

Also, seems to make sense to not list all the salted hashes by name, just say instead that something is required for salted hashes.

**Comment by rameshbhojan-okta** [comment]
> seems best to delete those two column and put the information in the descriptions of the properties.

Sounds good... Alternatively add a column called `NOTE` or `comments`, and mention those special requirements about min/max length, or min/max values for respective rows, where applicable.

**Comment by motikagan-okta** [comment]
Have deleted the columns, and put the information in the "Description" column.
Screen Shot 2020-04-07 at 5 07 09 PM

File: packages/@okta/vuepress-site/docs/reference/api/users/index.md [hunk 1]
```diff
@@ -3987,13 +3987,13 @@ Specifies a hashed password to import into Okta. This allows an existing passwor
...
+| -------- | ----- | ---------- |
+| algorithm  | String   | The algorithm used to generate the hash using the password (and salt, when applicable). Must be set to BCRYPT, SHA-512, SHA-256, SHA-1 or MD5. |
+| value      | String   | For SHA-512, SHA-256, SHA-1, MD5, This is the actual base64-encoded hash of the password (and salt, if used). This is the Base64 encoded `value` of the SHA-512/SHA-256/SHA-1/MD5 digest that was computed by either pre-fixing or post-fixing the `salt` to the `password`, depending on the `saltOrder`. If a `salt` was not used in the `source` system, then this should just be the the Base64 encoded `value` of the password's SHA-512/SHA-256/SHA-1/MD5 digest. For BCRYPT, This is the actual radix64-encoded hashed password. |
+| salt       | String   | For salted algorithms, specifies the base64-encoded password salt used to generate the hash. For BCRYPT, specifies the radix64-encoded password salt used to generate the hash. Only required for salted algorithms. Length must be 22. |
+| workFactor | Integer  | Governs the strength of the hash and the time required to compute it. Only required for BCRYPT algorithm. Minimum value is 1, and maximum is 20. |
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/reference/api/users/index.md:3995]:
```suggestion
| workFactor | Number  | Governs the strength of the hash and the time required to compute it. Only required for BCRYPT algorithm. Minimum value is 1, and maximum is 20. |
```

"Integer" is not a valid JSON data type.

**Pull request merged**